sherpa/astro/io/crates_backend.py

```diff
@@ -934,13 +934,13 @@ def get_pha_data(arg, make_copy=True, use_background=False):
 
     if pha is None:
         pha = phadataset.get_crate(phadataset.get_current_crate())
-        if (pha.get_key('HDUCLAS1').value == 'SPECTRUM' or
-                pha.get_key('HDUCLAS2').value == 'SPECTRUM'):
+        if (pha.get_key('HDUCLAS1').value == b'SPECTRUM' or
```

Comment:
Eek. Is crates really returning `bytes` rather than `str` here? That doesn't seem sensible (the domain is a better map to the FITS standard than unicode, but it's not perfect, and I think it's likely to confuse people). I'll have to see what astropy.io.fits does.

Would it be worth changing these `<crate>.get_key(...).value` calls to use `_try_key` and get it to convert from bytes to str, so that if crates changes behavior we (may) have less places to change?